File: cash.c
#include <cs50.h>
#include <stdio.h>

int calculate_quarters(int cents);
int calculate_dimes(int cents);
int calculate_nickels(int cents);
int calculate_pennies(int cents);
int get_cents(void);

int main(void)
{
    // Prompt the user for change owed, in cents
    int cents = get_cents();

    // Calculate quarters
    int quarters = calculate_quarters(cents);
    cents = cents - (quarters * 25);

    // Calculate dimes
    int dimes = calculate_dimes(cents);
    cents = cents - (dimes * 10);

    // Calculate nickels
    int nickels = calculate_nickels(cents);
    cents = cents - (nickels * 5);

    // Calculate pennies
    int pennies = calculate_pennies(cents);
    cents = cents - (pennies * 1);

    // Sum and print

    int coins = quarters + dimes + nickels + pennies;

    printf("%i\n", coins);
}
// Prompt cents

int get_cents(void)
{

    int cents;

    do
    {
        cents = get_int("Change owed: ");
    }
    while (cents < 0);
    return cents;
}

// Prompt quarters
int calculate_quarters(int cents)
{
    return cents / 25;
}
// Prompt dimes
int calculate_dimes(int cents)
{
    return cents / 10;
}
// Prompt nickels
int calculate_nickels(int cents)
{
    return cents / 5;
}
// Prompt pennies
int calculate_pennies(int cents)
{
    return cents / 1;
}

File: sentimental-cash/cash.py
from cs50 import get_float


def main():
    # Get user's input
    while True:
        money = get_float("Change: ")
        if money <= 0:
            pass
        else:
            money = money * 100
            break

    # Calculate quarters
    quarters = calcQuarquers(money)
    money = money % 25

    # Calculate dimes
    dimes = calcDimes(money)
    money = money % 10

    # Calculate nickels
    nickels = calcNickels(money)
    money = money % 5

    # Calculate pennies
    pennies = calcPennies(money)
    money = money % 1

    # Sum and print
    change = quarters + dimes + nickels + pennies
    print(int(change))


def calcQuarquers(money):
    return money // 25


def calcDimes(money):
    return money // 10


def calcNickels(money):
    return money // 5


def calcPennies(money):
    return money // 1


main()
